bot/exts/fun/duck_pond.py

```diff
@@ -47,14 +47,18 @@ def is_staff(member: MemberOrUser) -> bool:
         return False
 
     @staticmethod
-    def is_helper_viewable(channel: TextChannel) -> bool:
+    def is_helper_viewable(channel: Union[TextChannel, Thread]) -> bool:
```

Comment:
`permissions_for` works on all `GuildChannel` objects, and also takes into account default role permissions, so this whole function can be simplified quite a lot. The `channel` type hint can also be changed to `discord.GuildChannel`, since this funciton is available on all GuildChannel objects.

```py
def is_helper_viewable(channel: discord.GuildChannel) -> bool:
"""Return whether helpers can view the given channel."""
    helper_role = guild.get_role(constants.Roles.helpers)
    return channel.permissions_for(helper_role).view_channel
```

Reply:
Although, the only channel you can get reactions from is `TextChannel` and `Thread` (also `PrivateChannel`, but we do a check for the guild), so the current typehint is also correct.

Reply:
Yea, that's true. Although looking at the file, this function is only actually used once. So imo this function can be deleted and [this condition](https://github.com/python-discord/bot/blob/main/bot/exts/fun/duck_pond.py#L172-L174) can just be changed to
```py
helper_role = guild.get_role(constants.Roles.helpers)
if not channel.permissions_for(helper_role).view_channel:
    return
```

Reply:
@TizzySaurus I am looking through my notifications and found this PR, are you looking to implement the feedback Chris mentioned here?

Reply:
This is resolved in https://github.com/python-discord/bot/pull/2087/commits/532ce7d7785103c8c67341d8d24d5301f76a8257.